### PR TITLE
remove special code for python 2.7 (six.PY2)

### PR DESCRIPTION
**Pull request by Sebastian-Lemling**
working on #2128.

### CODE REVIEW

File: tests/test_api_auth.py [hunk 1]
```diff
@@ -137,10 +136,6 @@ def test_01_auth_with_split(self):
                 self.assertEqual(401, res.status_code, res)
                 result = res.json.get("result")
                 self.assertFalse(result.get("status"), result)
-            if six.PY2:
-                expected = "The user User(login=u'cornelius@unknown', " \
-                           "realm=u'realm1', resolver='') exists in NO resolver."
-            else:
                 expected = "The user User(login='cornelius@unknown', " \
```
Comment (jona-samuel) [inline tests/test_api_auth.py:139]: remove the init of the line, please

like this
~~~Python
with self.app.test_request_context('/auth',
                                               method='POST',
                                               data={"username": "cornelius@unknown",
                                                     "realm": "realm1",
                                                     "password": "test"}):
                res = self.app.full_dispatch_request()
                self.assertEqual(401, res.status_code, res)
                result = res.json.get("result")
                self.assertFalse(result.get("status"), result)
expected = "The user User(login='cornelius@unknown', " \
~~~

File: tests/test_api_auth.py [hunk 2]
```diff
@@ -189,10 +184,6 @@ def test_01_auth_with_split(self):
                 self.assertEqual('Authentication failure. Wrong credentials',
                                  result['error']['message'], result)
             # the realm will be split from the login name
-            if six.PY2:
-                expected = "The user User(login=u'selfservice', " \
-                           "realm=u'realm3', resolver='') exists in NO resolver."
-            else:
                 expected = "The user User(login='selfservice', " \
```
Comment (jona-samuel) [inline tests/test_api_auth.py:187]: here also

File: tests/test_api_auth.py [hunk 3]
```diff
@@ -238,10 +229,6 @@ def test_02_auth_without_split(self):
                 self.assertEqual(4031, result['error']['code'], result)
                 self.assertEqual('Authentication failure. Wrong credentials',
                                  result['error']['message'], result)
-            if six.PY2:
-                expected = "The user User(login=u'cornelius@realm1', " \
-                           "realm=u'realm1', resolver='') exists in NO resolver."
-            else:
                 expected = "The user User(login='cornelius@realm1', " \
```
Comment (jona-samuel) [inline tests/test_api_auth.py:232]: here also

File: tests/test_api_auth.py [hunk 4]
```diff
@@ -312,10 +299,6 @@ def test_02_auth_without_split(self):
                 self.assertEqual(401, res.status_code, res)
                 result = res.json.get("result")
                 self.assertFalse(result.get("status"), result)
-            if six.PY2:
-                expected = "The user User(login=u'cornelius@unknown', " \
-                           "realm=u'realm1', resolver='') exists in NO resolver."
-            else:
                 expected = "The user User(login='cornelius@unknown', " \
```
Comment (jona-samuel) [inline tests/test_api_auth.py:302]: here also

File: tests/test_api_auth.py [hunk 5]
```diff
@@ -348,10 +331,6 @@ def test_02_auth_without_split(self):
                 result = res.json.get("result")
                 self.assertFalse(result.get("status"), result)
                 self.assertEqual(4031, result['error']['code'], result)
-            if six.PY2:
-                expected = "The user User(login=u'selfservice@realm3', " \
-                           "realm=u'realm1', resolver='') exists in NO resolver."
-            else:
                 expected = "The user User(login='selfservice@realm3', " \
```
Comment (jona-samuel) [inline tests/test_api_auth.py:334]: here also

File: tests/test_api_auth.py [hunk 6]
```diff
@@ -370,11 +349,7 @@ def test_02_auth_without_split(self):
                 self.assertEqual(4031, result['error']['code'], result)
                 self.assertEqual('Authentication failure. Wrong credentials',
                                  result['error']['message'], result)
-            # the realm will be split from the login name
-            if six.PY2:
-                expected = "The user User(login=u'selfservice@realm1', " \
-                           "realm=u'realm3', resolver='') exists in NO resolver."
-            else:
+                # the realm will be split from the login name
                 expected = "The user User(login='selfservice@realm1', " \
```
Comment (jona-samuel) [inline tests/test_api_auth.py:353]: here also

**Review by jona-samuel — CHANGES_REQUESTED** [review]

File: tests/test_api_auth.py (hunk 1 above)
Comment (NNLemling) [reply tests/test_api_auth.py]: fixed in a1e752d6c5abb37fab930e65b3c64b3680e3f3f1.

File: tests/test_api_auth.py (hunk 2 above)
Comment (NNLemling) [reply tests/test_api_auth.py]: fixed in a1e752d6c5abb37fab930e65b3c64b3680e3f3f1.

File: tests/test_api_auth.py (hunk 3 above)
Comment (NNLemling) [reply tests/test_api_auth.py]: fixed in a1e752d6c5abb37fab930e65b3c64b3680e3f3f1.

File: tests/test_api_auth.py (hunk 4 above)
Comment (NNLemling) [reply tests/test_api_auth.py]: fixed in a1e752d6c5abb37fab930e65b3c64b3680e3f3f1.

File: tests/test_api_auth.py (hunk 5 above)
Comment (NNLemling) [reply tests/test_api_auth.py]: fixed in a1e752d6c5abb37fab930e65b3c64b3680e3f3f1.

File: tests/test_api_auth.py (hunk 6 above)
Comment (NNLemling) [reply tests/test_api_auth.py]: fixed in a1e752d6c5abb37fab930e65b3c64b3680e3f3f1.

**Review by jona-samuel — APPROVED** [review]

**Pull request merged**